task/script.js

```diff
@@ -1,9 +1,285 @@
 const API_BASE_URL = window.API_BASE_URL;
+const SKILLS_API = 'https://services.realdevsquad.com/skilltree/v1/skills';
+
 const suggestedUsers = [];
 const allUser = [];
+
+const params = new URLSearchParams(window.location.search);
+const isDev = params.get('dev') === 'true';
+
+const featureUrlField = document
+  .getElementById('featureUrl')
+  .closest('.inputBox');
+
+if (isDev && featureUrlField) {
+  featureUrlField.style.display = 'none';
+}
+const featureRadio = document.getElementById('feature')?.closest('.inputBox');
+
+if (isDev && featureRadio) {
+  featureRadio.style.display = 'none';
+}
+const taskLevelDiv = document
+  .querySelector('.inputBox label[for="taskLevel"]')
+  .closest('.inputBox');
+
+if (isDev && taskLevelDiv) {
+  taskLevelDiv.style.display = 'none';
+}
+const skillComponentDiv = document
+  .querySelector('.multi-select-button')
+  ?.closest('.inputBox');
+if (!isDev && skillComponentDiv) {
+  skillComponentDiv.style.display = 'none';
+}
+class MultiSelect {
+  constructor(container) {
+    this.container = container;
+    this.selectedValues = new Set();
+    this.button = container.querySelector('.multi-select-button');
+    this.selectedItemsContainer = container.querySelector('.selected-items');
+    this.placeholder = container.querySelector('.placeholder');
+    this.popover = container.querySelector('.popover');
+    this.searchInput = container.querySelector('.search-input');
+    this.optionsList = container.querySelector('.options-list');
+    this.options = [];
+    this.focusedIndex = -1;
+
+    this.initializeSkills();
+    this.bindEvents();
+  }
+
+  async initializeSkills() {
+    try {
+      const skills = await this.fetchSkills();
+      this.options = skills.map((skill) => ({
+        value: skill.id.toString(),
+        label: skill.name,
+      }));
+      this.init();
+    } catch (error) {
+      alert(`Error Initializing skills: ${error}`);
+    }
+  }
+
+  async fetchSkills() {
+    if (!isDev) return [];
+    try {
+      const response = await fetch(SKILLS_API, {
+        credentials: 'include',
+      });
+      return await response.json();
+    } catch (error) {
+      alert(`Error fetching skills: ${error}`);
+      return [];
+    }
+  }
```

Comment:
- let's move this outside of the class as it is not a property of the class, also this will allow us to  reuse this method without instantiating the `MultiSelect` class.

Reply:
Done